openassessment/fileupload/backends/django_storage.py

```diff
@@ -44,6 +80,20 @@ def remove_file(self, key):
         Returns False if the file does not exist, and so was not removed.
         """
         path = self._get_file_path(key)
+        metadata_path = '{path}{ext}'.format(path=path, ext=self.METADATA_FILE_EXTENSION)
+
+        # Checks that .meta file exists for this key.
+        if default_storage.exists(metadata_path):
+            metadata_file = default_storage.open(metadata_path).read()
+            metadata_dict = json.loads(metadata_file)
```

Comment:
We could create a function to obtain `metadata_dict` then we could use it here, and also above in `get_download_url()`. We would be avoiding code repetition.

Reply:
Addresed @diegomillan